### PR TITLE
DB-363: Automatic private tab

### PR DESCRIPTION
**Pull request by maxim-cliqz**
The feature is not fully ready yet, but it's disabled by default, so we should be safe to merge it into master.
There's also a compile time dependency - a database of adult domains - which is not included to repository for the sake of it's performance. It's binary and makes little sense to me to store it here.

@chrmod @alver-cliqz you're welcome to review too

### CODE REVIEW

File: mozilla-release/browser/components/BloomFilterCompiler.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
+"use strict"
+// Copyright Cliqz GmbH, 2015.
```
Comment (luciancor) [inline mozilla-release/browser/components/BloomFilterCompiler.js]: @maxim-cliqz 2016 :)

File: mozilla-release/browser/components/BloomFilterCompiler.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  if (!check)
+    print(line, check);
+}
+print("===========");
+print(ok ? "OK" : "Checks for some lines failed!");
+
+}
+catch (e) {
+  fail(e);
+}
```
Comment (luciancor) [inline mozilla-release/browser/components/BloomFilterCompiler.js]: @maxim-cliqz is this file used offline?

File: mozilla-release/browser/config/cliqz.mozconfig [hunk 3]
```diff
@@ -13,3 +13,4 @@ ac_add_options --with-app-name=CLIQZ
 export MOZ_APP_PROFILE=CLIQZ
 export MOZ_AUTOMATION_UPLOAD=1
 export MOZ_SOURCE_REPO=https://github.com/cliqz-oss/browser-f
+#export CQZ_ADULT_DOMAINS_BF=$topsrcdir/porn-domains.bin
```
Comment (luciancor) [inline mozilla-release/browser/config/cliqz.mozconfig]: @maxim-cliqz can we please rename this to "adult-*"

Comment (maxim-cliqz) [inline mozilla-release/browser/config/cliqz.mozconfig]: ah, this line should be just removed, it's an overlooked leftover

File: mozilla-release/browser/components/BloomFilterCompiler.js (hunk 2 above)
Comment (maxim-cliqz) [inline mozilla-release/browser/components/BloomFilterCompiler.js]: yes, it's just for internal usage, to prepare the database
suggest a better place, and I'll move it :)

**Comment by maxim-cliqz** [comment]
retest this please

**Comment by maxim-cliqz** [comment]
retest this please

**Comment by maxim-cliqz** [comment]
retest this please

**Comment by maxim-cliqz** [comment]
retest this please

**Pull request merged**